platform/FileHandle.h

```diff
@@ -152,8 +162,106 @@ class FileHandle {
      */
     MBED_DEPRECATED_SINCE("mbed-os-5.4", "Replaced by FileHandle::size")
     virtual off_t flen() { return size(); }
+
+    /** Set blocking or non-blocking mode of the file operation like read/write.
+     *  Definition depends upon the subclass implementing FileHandle.
+     *  The default is blocking.
+     *
+     *  @param blocking true for blocking mode, false for non-blocking mode.
+     */
+    virtual int set_blocking(bool blocking) {
+        return -1;
+    }
+
+    /** Check for poll event flags
+     * The input parameter can be used or ignored - the could always return all events,
+     * or could check just the events listed in events.
+     * Call is non-blocking - returns instantaneous state of events.
+     * Whenever an event occurs, the derived class must call _poll_change().
+     * @param events bitmask of poll events we're interested in - POLLIN/POLLOUT etc.
+     * @return bitmask of poll events that have occurred.
+     */
+    virtual short poll(short events) const {
```

Comment:
Quick notes:

- `mbed_poll.h` seems like the best path forward

- I'm not sure about `mbed::poll` vs `mbed_poll`, so I'll leave that up to you (we do already have mbed::callback).

- I would still push for uint16_t (or int16_t if you want), it will be the same underlying type as short, just with an explicit width

- I copied chunks of that description from the linux manual I think. It never hurts to document more than necessary. Related thought: should we just add links to the open group docs on poll/sigio?

---

My thinking is that the "state fetching" of `FileHandle::poll` can turn a non-definite sigio into a definite sigio.

`FileHandle::poll` is a harder requirement but I believe this is necessary for the "definite" functionality users want (poll, event driven). I think at some point it will need to be added to the socket API.

I'm fairly confident you can move `_poll_change` out of the FileHandle class. Just let the implementors handle the callback. Here's a quick poll based around this idea for sanity:
https://gist.github.com/geky/7eee6937f3d55552d4e365bf02de114f